doc/nrf/ug_bootloader.rst

```diff
@@ -92,7 +92,96 @@ These signed variants can be used to perform an upgrade procedure through the :r
 Adding a bootloader chain to your application
 *********************************************
 
-See :ref:`bootloader_build_and_run` in the documentation of the :ref:`bootloader` sample for instructions on how to add an immutable bootloader as first stage of the secure bootloader chain.
+The |NCS| includes a sample implementation of an immutable bootloader.
+As upgradable bootloader, it provides and extends :doc:`mcuboot:index`.
 
-See :doc:`mcuboot:index` for information on how to implement an upgradable bootlader.
+Both bootloaders can easily be included in your application using :ref:`ug_multi_image`.
+
+Adding the immutable bootloader
+==============================
+
+To add an immutable bootloader to your application, set :option:`CONFIG_SECURE_BOOT` and add your private key file under :option:`CONFIG_SB_SIGNING_PUBLIC_KEY`.
+|how_to_configure|
+
+See the documentation of the :ref:`bootloader` sample for more information.
+The :ref:`bootloader_build_and_run` section has detailed instructions for adding the immutable bootloader as first stage of the secure bootloader chain.
+
+Adding MCUboot as an upgradable bootloader
+===============================
+
+To add an upgradable bootloader in this case MCUBoot to your applications, set :option:`CONFIG_BOOTLOADER_MCUBOOT`.
+|how_to_configure|
+
+If you want an upgradable bootloader you should always include the immutable bootloader as well when adding MCUBoot as an upgradable bootloader.
+
+.. note::
+   It is possible to include this bootloader without the immutable bootloader.
+   In this case MCUboot will act as an immutable bootloader.
+
+   In the default configuration, most |NCS| samples include the upgradable bootloader but not the immutable bootloader.
+   The reason for this is that the immutable bootloader must be configured with a private key before use.
+
+The |NCS| uses MCUboot as upgradable bootloader.
+See :doc:`mcuboot:index` for information about the default implementation.
```

Comment:
```suggestion
See :doc:`mcuboot:index` for information about the default implementation of MCUboot.
```

Reply:
Why "default"? How do you get the default, and when does it stop being default (i.e. when do the linked docs stop being relevant)?

Reply:
Well, the thing is that the linked docs should ALWAYS be relevant, but we haven't started talking about how any of this works until the release freeze ...
I can't answer that question, and I doubt we'll fix the documentation of our MCUboot fork this week.

Reply:
Ok, I would suggest removing "default" then.

Reply:
"See :doc:`mcuboot:index` for the official MCUboot documentation."

Reply:
No. :( The official MCUboot documentation is on https://juullabs-oss.github.io/mcuboot/, ours is adapted a little bit, but not fully. We have taken out some topics, but not updated the ones that are included (for example, https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/mcuboot/readme-zephyr.html says that "Currently, the two image slots must be contiguous.", which is clearly wrong if one of them is in external flash).

Reply:
Ok, my questions about "default" still stand, but I can live with the sentence as is :).

Reply:
I'm not saying this is good, btw. There should definitely be a task to fix the MCUboot documentation - but this cannot be done the week before the release (which is exactly what happened last time as well ...)